lib/joke.rb

```diff
@@ -1,3 +1,28 @@
 class Joke
 
+  attr_reader :id, :question, :answer
+
+  # def initialize(joke_hash = {id: 1, question: "funny q", answer: "punchline"})
+  #
+  #   @id = joke_hash[id]
+  #   @question = joke_hash[question]
+  #   @answer = joke_hash[answer]
+  # end
```

Comment:
This first set here would have worked well.  The hash comes in as a single argument. So your initialize might look something like this: 

```
def initialize(joke_info) 
      @id = joke_info[:id]
      @question = joke_info[:question]
      @answer = joke_info[:answer]
end

Reply:
What you had may no have been working due to the fact that the hash getting passed in has symbols as the keys while you were calling keys like this `id`.

Reply:
In the first 10 minutes I had worked through tests and had code and tests that passed except for answer. I couldn't figure it out and started changing all sorts of other things. I don't remember what I started with.